cmd/postgres_exporter/queries.go

```diff
@@ -227,6 +229,9 @@ func parseUserQueries(content []byte) (map[string]intermediateMetricMap, map[str
 				master:         specs.Master,
 				cacheSeconds:   specs.CacheSeconds,
 			}
+			if len(specs.Databases) > 0 {
+				metricMap.databases = regexp.MustCompile(specs.Databases)
```

Comment:
@WnP are you right to tweak that? If not I'm happy to pick this up to get it merged.

Reply:
As far as I can see `regex.MustCompile` returns a `*Regexp` not error, also cannot see which panic you're talking about.

`parseUserQueries` return an error which is returned by `addQueries` which is then logged by `main.checkMapVersions`, still no panic there.

There's only few panic calls in the code base and they're in pg_settings.go `metric` function, which I did not modified.

Can you clarify what you want @sysadmind?

I'll be happy to rebase and fix the conflicts afterward.

Reply:
`regexp.MustCompile` will panic if there is an error parsing the regex. https://pkg.go.dev/regexp#MustCompile

`regexp.Compile` is the preferred option to avoid a panic.

Reply:
Actually, at this point we may want to hold off. queries.yaml has been deprecated and all existing queries have been ported into the collectors package.

Reply:
We're going to cut one more release with the currently supported features before we completely remove the yaml query functionality.

Reply:
Oh, sorry, my bad @sysadmind, completely miss that... was coded a long time ago :smile:.

Will database filter be available in the collector package?